### ISSUE TITLE
NetCDF Chunksize Error

### ISSUE THREAD

**Issue by dwhswenson** [openpathsampling/openpathsampling#660]
It appears that a new error will be introduced when we move to later versions of netCDF. Because we're currently pinned to avoid a `conda`-based issue (https://github.com/ContinuumIO/anaconda-issues/issues/1360), our tests are using:

* `libnetcdf 4.3.3.1-3`
* `netcdf4 1.2.2-np110py27_0`

On my local machine, I'm using:

* `libnetcdf 4.4.1_2` (called `netcdf` on MacPorts)
* `netcdf4 1.2.3_1` (called `py27-netcdf4` on MacPorts)

This said, I get the following error:

```python traceback
======================================================================
ERROR: testcollectivevariable.test_FunctionCV.test_storage_cv_function
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/dwhs/Dropbox/msm-tis/openpathsampling/tests/testcollectivevariable.py", line 125, in test_storage_cv_function
    storage_w.save(cv1)
  File "/Users/dwhs/Dropbox/msm-tis/openpathsampling/netcdfplus/netcdfplus.py", line 587, in save
    return store, store_idx, store.save(obj, idx)
  File "/Users/dwhs/Dropbox/msm-tis/openpathsampling/netcdfplus/stores/named.py", line 442, in save
    reference = super(UniqueNamedObjectStore, self).save(obj, idx)
  File "/Users/dwhs/Dropbox/msm-tis/openpathsampling/netcdfplus/stores/named.py", line 254, in save
    reference = super(NamedObjectStore, self).save(obj)
  File "/Users/dwhs/Dropbox/msm-tis/openpathsampling/netcdfplus/stores/object.py", line 875, in save
    self._save(obj, n_idx)
  File "/Users/dwhs/Dropbox/msm-tis/openpathsampling/storage/stores/collectivevariable.py", line 18, in _save
    self.add_diskcache(cv)
  File "/Users/dwhs/Dropbox/msm-tis/openpathsampling/storage/stores/collectivevariable.py", line 96, in add_diskcache
    chunksize=chunksize
  File "/Users/dwhs/Dropbox/msm-tis/openpathsampling/storage/stores/snapshot_wrapper.py", line 673, in add_cv
    simtk_unit=params['simtk_unit'],
  File "/Users/dwhs/Dropbox/msm-tis/openpathsampling/netcdfplus/netcdfplus.py", line 1145, in create_variable
    var_name, nc_type, dimensions, chunksizes=chunksizes,
  File "netCDF4/_netCDF4.pyx", line 2222, in netCDF4._netCDF4.Dataset.createVariable (netCDF4/_netCDF4.c:17109)
  File "netCDF4/_netCDF4.pyx", line 3181, in netCDF4._netCDF4.Variable.__init__ (netCDF4/_netCDF4.c:29323)
RuntimeError: NetCDF: Bad chunk sizes.
```

Note that, from the entire test suite, the only error is in this one test. Did something get updated in other parts of the code, but not in that one?

**Comment by jhprinz** [comment]
I will look into this. I am still using 4.3.3.1 because after that I also get an error. It could be that netcdf changed some restrictions on the chunksizes but so far I have not been able to understand how chunksizes work with variable length entries.

**Comment by jhprinz** [comment]
Found a fix. Will make a quick fix, otherwise #662 contains a fix also. Seems there is a bug in the chunk size handling and you need to change it `1` once you add a new variable and objects in associated dimensions have been written.

**Comment by jhprinz** [comment]
Resolved by #673 .
